benchmark/CMakeLists.txt

```diff
@@ -50,15 +50,23 @@ macro(add_dataset datasets dataset path_to_template range)
 endmacro()
 
 add_custom_target(benchmarks)
-macro(add_chart target)
+macro(add_chart type case)
     cmake_parse_arguments(ARGS "" "" "DATASETS" ${ARGN})
 
+    set(type_target "benchmarks.${type}")
+    set(case_target "${type_target}.${case}")
+
+    if (NOT TARGET ${type_target})
+        add_custom_target(${type_target})
+    endif()
+
     # If we have no datasets, perhaps they were all disabled in the current
     # configuration. In this case, we omit the benchmark instead of failing.
     list(LENGTH ARGS_DATASETS _n_datasets)
     if (_n_datasets GREATER 0)
-        metabench_add_chart(${target} ${ARGN} OUTPUT index.html)
-        add_dependencies(benchmarks ${target})
+        metabench_add_chart(${case_target} ${ARGN} OUTPUT index.html)
+        add_dependencies(${type_target} ${case_target})
+        add_dependencies(benchmarks ${type_target})
```

Comment:
What happens when we call `add_dependencies` multiple times on the same targets? Here, we'll call `add_dependencies(benchmarks benchmarks.hetero)` and `add_dependencies(benchmarks benchmarks.type)` multiple times.